lib/dockerregistry/inventory.go

```diff
@@ -1639,11 +1650,11 @@ func (sc *SyncContext) ExecRequests(
 				(*mutex).Lock()
 				err = fmt.Errorf("Encountered an error during the" +
 					" promotion step")
+				sc.Logs.Errors = append(sc.Logs.Errors, reqRes.Errors...)
 				(*mutex).Unlock()
 				klog.Errorf(
 					"Request %v: error(s) encountered: %v\n",
-					reqRes.Context,
-					reqRes.Errors)
+					reqRes.Context, reqRes.Errors)
```

Comment:
This change isn't necessary. Let's try to keep the diff clean...!

Reply:
The way I would do this is: squash all 6 commits into 1, then use `git reset ...` to let git perform the "undo" for me.

Reply:
Undo?